cmd/config-sync/config_sync.go

```diff
@@ -249,23 +255,36 @@ func syncRouterConfig(agent *qdr.Agent, desired *qdr.RouterConfig, c *ConfigSync
 	}
 }
 
-func syncSecrets(configSync *ConfigSync, changes *qdr.BridgeConfigDifference, sharedTlsFilesDir string) error {
-	for _, added := range changes.HttpListeners.Added {
-		if len(added.SslProfile) > 0 {
-			log.Printf("Copying cert files related to HTTP Listener sslProfile %s", added.SslProfile)
-			err := configSync.copyCertsFilesToPath(sharedTlsFilesDir, added.SslProfile, added.SslProfile)
+func syncSecrets(routerConfig *qdr.RouterConfig, changes *qdr.BridgeConfigDifference, sharedPath string, copyCerts func(string, string, string) error, newSSlProfile func(profile qdr.SslProfile) error, delSslProfile func(string, string) error) error {
```

Comment:
Just a cosmetic thing (so feel free to ignore it), but maybe the last 3
arguments could be represented by a type, or eventually, by an interface.

```
type CopyCerts func(string, string, string) error
type NewSSlProfile func(profile qdr.SslProfile) error
type DelSslProfile func(string, string) error
```

or

```
type SslProfileManager interface {
	Create(profile qdr.SslProfile) error
	Delete(name string) error
	Copy(path string, profilename string, secretname string) error
}
```

Reply:
I like it! it is way better. Thanks!